cmd/rqlite/main.go

```diff
@@ -93,6 +95,12 @@ func main() {
 		}
 		term.Close()
 
+		hosts := createHostList(argv)
+		client := httpcl.NewClient(httpClient, hosts,
+			httpcl.WithScheme(argv.Protocol),
+			httpcl.WithBasicAuth(argv.Credentials),
```

Comment:
With....do you write Java by any chance? :-)

Reply:
AFAIK this is a pretty common Go pattern for writing `Builder` like structures, where you don't want to write a constructor for every possible combination of arguments.

Since I assumed the client can be used elsewhere, I went with this to be more flexible.

Reply:
And Yes, I write Java from time to time :-)

Reply:
I haven't come across in the Go code I work with, though I recognize it. I'm fine with here.